nova/virt/vmwareapi/vmops.py

```diff
@@ -2014,10 +2007,47 @@ def finish_migration(self, context, migration, instance, disk_info,
         """Completes a resize, turning on the migrated instance."""
         vm_ref = vm_util.get_vm_ref(self._session, instance)
 
-        # 5. Update ephemerals if necessary
+        flavor = instance.flavor
+        boot_from_volume = compute_utils.is_volume_backed_instance(context,
+                                                                   instance)
+        reattach_volumes = False
+        # 2. Relocate the VM if necessary
+        # If the dest_compute is different from the source_compute, it means we
+        # need to relocate the VM here since we are running on the dest_compute
+        if migration.source_compute != migration.dest_compute:
+            self._detach_volumes(instance, block_device_info)
+            reattach_volumes = True
+            LOG.debug("Relocating VM for migration to %s",
+                      migration.dest_compute, instance=instance)
```

Comment:
I think it would be nice to have a corresponding `Relocated VM` down the line, so we can get some timers or see if it broke in the middle.

Reply:
Same above for the other relocate.

Reply:
Added below.